### ISSUE TITLE
Xcode 16 issue

### ISSUE THREAD

**Issue by Ramprasathselvam** [google/android-auto-companion-ios#6]
Android-auto-companion-ios is not working in Xcode 16 alone.

After adding Android-auto-companion-ios package into our existing iOS Applications the app is not running in Xcode 16 alone, Xcode 15.x series it's working fine.

```
App installation failed: Unable to Install “Connect”
Domain: IXUserPresentableErrorDomain
Code: 1
Failure Reason: Please try again later.
Recovery Suggestion: Failed to load Info.plist from bundle at path /Users/ramprasathselvam/Library/Developer/CoreSimulator/Devices/A39595A6-83B4-4D20-B6F6-064A63E43501/data/Library/Caches/com.apple.mobile.installd.staging/temp.E2znvR/extracted/NCI.app/Frameworks/AndroidAutoUKey2Wrapper.framework; Extra info about "/Users/ramprasathselvam/Library/Developer/CoreSimulator/Devices/A39595A6-83B4-4D20-B6F6-064A63E43501/data/Library/Caches/com.apple.mobile.installd.staging/temp.E2znvR/extracted/NCI.app/Frameworks/AndroidAutoUKey2Wrapper.framework/Info.plist": Couldn't stat /Users/ramprasathselvam/Library/Developer/CoreSimulator/Devices/A39595A6-83B4-4D20-B6F6-064A63E43501/data/Library/Caches/com.apple.mobile.installd.staging/temp.E2znvR/extracted/NCI.app/Frameworks/AndroidAutoUKey2Wrapper.framework/Info.plist: No such file or directory
User Info: {
    DVTErrorCreationDateKey = "2024-09-18 13:07:49 +0000";
    IDERunOperationFailingWorker = "_IDEInstalliPhoneSimulatorWorker";
    SimCallingSelector = "installApplication:withOptions:error:";
}
--
Unable to Install “Connect”
Domain: IXUserPresentableErrorDomain
Code: 1
Failure Reason: Please try again later.
Recovery Suggestion: Failed to load Info.plist from bundle at path /Users/ramprasathselvam/Library/Developer/CoreSimulator/Devices/A39595A6-83B4-4D20-B6F6-064A63E43501/data/Library/Caches/com.apple.mobile.installd.staging/temp.E2znvR/extracted/NCI.app/Frameworks/AndroidAutoUKey2Wrapper.framework; Extra info about "/Users/ramprasathselvam/Library/Developer/CoreSimulator/Devices/A39595A6-83B4-4D20-B6F6-064A63E43501/data/Library/Caches/com.apple.mobile.installd.staging/temp.E2znvR/extracted/NCI.app/Frameworks/AndroidAutoUKey2Wrapper.framework/Info.plist": Couldn't stat /Users/ramprasathselvam/Library/Developer/CoreSimulator/Devices/A39595A6-83B4-4D20-B6F6-064A63E43501/data/Library/Caches/com.apple.mobile.installd.staging/temp.E2znvR/extracted/NCI.app/Frameworks/AndroidAutoUKey2Wrapper.framework/Info.plist: No such file or directory
--
Failed to load Info.plist from bundle at path /Users/ramprasathselvam/Library/Developer/CoreSimulator/Devices/A39595A6-83B4-4D20-B6F6-064A63E43501/data/Library/Caches/com.apple.mobile.installd.staging/temp.E2znvR/extracted/NCI.app/Frameworks/AndroidAutoUKey2Wrapper.framework; Extra info about "/Users/ramprasathselvam/Library/Developer/CoreSimulator/Devices/A39595A6-83B4-4D20-B6F6-064A63E43501/data/Library/Caches/com.apple.mobile.installd.staging/temp.E2znvR/extracted/NCI.app/Frameworks/AndroidAutoUKey2Wrapper.framework/Info.plist": Couldn't stat /Users/ramprasathselvam/Library/Developer/CoreSimulator/Devices/A39595A6-83B4-4D20-B6F6-064A63E43501/data/Library/Caches/com.apple.mobile.installd.staging/temp.E2znvR/extracted/NCI.app/Frameworks/AndroidAutoUKey2Wrapper.framework/Info.plist: No such file or directory
Domain: MIInstallerErrorDomain
Code: 35
User Info: {
    FunctionName = "-[MIBundle _validateWithError:]";
    LegacyErrorString = PackageInspectionFailed;
    SourceFileLine = 64;
}
--

Event Metadata: com.apple.dt.IDERunOperationWorkerFinished : {
    "device_identifier" = "C1BA73C6-D5D7-40A3-A3C7-30E22E0730E5";
    "device_model" = "Watch7,4";
    "device_osBuild" = "11.0 (22R349)";
    "device_platform" = "com.apple.platform.watchsimulator";
    "device_thinningType" = "Watch7,4";
    "dvt_coredevice_version" = "397.21";
    "dvt_coresimulator_version" = 987;
    "dvt_mobiledevice_version" = "1759.2.1";
    "launchSession_schemeCommand" = Run;
    "launchSession_state" = 1;
    "launchSession_targetArch" = "x86_64";
    "operation_duration_ms" = 5516;
    "operation_errorCode" = 1;
    "operation_errorDomain" = IXUserPresentableErrorDomain;
    "operation_errorWorker" = "_IDEInstalliPhoneSimulatorWorker";
    "operation_name" = IDERunOperationWorkerGroup;
    "param_debugger_attachToExtensions" = 0;
    "param_debugger_attachToXPC" = 1;
    "param_debugger_type" = 3;
    "param_destination_isProxy" = 1;
    "param_destination_platform" = "com.apple.platform.watchsimulator";
    "param_diag_113575882_enable" = 0;
    "param_diag_MainThreadChecker_stopOnIssue" = 0;
    "param_diag_MallocStackLogging_enableDuringAttach" = 0;
    "param_diag_MallocStackLogging_enableForXPC" = 1;
    "param_diag_allowLocationSimulation" = 1;
    "param_diag_checker_tpc_enable" = 1;
    "param_diag_gpu_frameCapture_enable" = 0;
    "param_diag_gpu_shaderValidation_enable" = 0;
    "param_diag_gpu_validation_enable" = 0;
    "param_diag_guardMalloc_enable" = 0;
    "param_diag_memoryGraphOnResourceException" = 0;
    "param_diag_mtc_enable" = 1;
    "param_diag_queueDebugging_enable" = 1;
    "param_diag_runtimeProfile_generate" = 0;
    "param_diag_sanitizer_asan_enable" = 0;
    "param_diag_sanitizer_tsan_enable" = 0;
    "param_diag_sanitizer_tsan_stopOnIssue" = 0;
    "param_diag_sanitizer_ubsan_enable" = 0;
    "param_diag_sanitizer_ubsan_stopOnIssue" = 0;
    "param_diag_showNonLocalizedStrings" = 0;
    "param_diag_viewDebugging_enabled" = 1;
    "param_diag_viewDebugging_insertDylibOnLaunch" = 1;
    "param_install_style" = 2;
    "param_launcher_UID" = 2;
    "param_launcher_allowDeviceSensorReplayData" = 0;
    "param_launcher_kind" = 0;
    "param_launcher_style" = 0;
    "param_launcher_substyle" = 8192;
    "param_runnable_appExtensionHostRunMode" = 0;
    "param_runnable_productType" = "com.apple.product-type.application";
    "param_structuredConsoleMode" = 1;
    "param_testing_launchedForTesting" = 0;
    "param_testing_suppressSimulatorApp" = 0;
    "param_testing_usingCLI" = 0;
    "sdk_canonicalName" = "iphonesimulator18.0";
    "sdk_osVersion" = "18.0";
    "sdk_variant" = iphonesimulator;
}
--


System Information

macOS Version 14.5 (Build 23F79)
Xcode 16.0 (23051) (Build 16A242d)
Timestamp: 2024-09-18T18:37:49+05:30
```

**Comment by tapsns** [comment]
Thank you for posting the issue. It is a known issue that Companion currently builds with Xcode 15.x but not Xcode 16 though I haven't seen that specific issue with info.plist. 

We have a new version in development that builds with both Xcode 15 and 16. This new version will be available with the upcoming Companion release and includes support for Swift 6 among other changes.

**Comment by Ramprasathselvam** [comment]
Hi @tapsns, Thanks for the response, When will the updated version be made available to the public? We may schedule our delivery depending on that.

**Comment by tapsns** [comment]
Hi Ramprasath,

I would estimate November. Internally, we have the changes, but we need to sync with changes for the other platforms and do QA testing, etc.

I've gone ahead an pushed the changes to the `preview` branch. Let me know if that fixes the problem. Thanks.

Best regards,
Tom

**Comment by tapsns** [comment]
I released version Companion SDK version 4.2.1 which allows it to build with Xcode 16.

**Comment by Ramprasathselvam** [comment]
This Xcode 16 issue has been resolved. Thanks for the support🤗.
